docs/modules/third_party_rules_callbacks.md

```diff
@@ -265,6 +265,33 @@ server_.
 
 If multiple modules implement this callback, Synapse runs them all in order.
 
+### `unbind_threepid`
+
+_First introduced in Synapse v1.74.0_
+
+```python
+async def unbind_threepid(
+    user_id: str, medium: str, address: str, identity_server: str
+) -> Tuple[bool, bool]:
+```
+
+Called before a threepid association is removed.
+
+The module is given the Matrix ID of the user to which an association is to be removed,
+as well as the medium (`email` or `msisdn`), address of the third-party identifier and
+the identity server where the threepid was successfully registered.
+
+A module can hence do its own custom unbinding, if for example it did also registered a custom
+binding logic with `on_threepid_bind`.
+
+It should return a tuple of 2 booleans:
+- first one should be `True` on a success calling the identity server, otherwise `False` if
+the identity server doesn't support unbinding (or no identity server found to contact).
+- second one should be `True` if unbind needs to stop there. In this case no other module
+unbind will be called, and the default unbind made to the IS that was used on bind will also be
+skipped. In any case the mapping will be removed from the Synapse 3pid remote table,
+except if an Exception was raised at some point.
```

Comment:
```suggestion
- whether the unbind was successful (`True` in this case, `False` otherwise).
- whether Synapse should stop the unbinding process there (`True` in this case, `False` otherwise). In
this case, Synapse will also remove any local record of an association between the local user and the
third-party identifier on a remote identity server.

If multiple modules implement this callback, they will be considered in order. Unless the return value of
a callback indicates to stop the unbinding process, Synapse falls through to the next one. Otherwise,
Synapse will not call any subsequent implementation of this callback.
```

We should also add to the last paragraph a mention of how the first boolean is processed once we have resolved this question in a previous thread.

A few things this suggestion changes:
* I don't believe it is safe to assume all modules will do their binding/unbinding against an identity server, so it might not make sense to tie the definition of return values to the existence of an identity server.
* The bit about exceptions don't feel right to me: first, looking at the code it doesn't look correct; second the way it's phrased makes it sound like modules should use exceptions to define whether Synapse should delete the remote record, and using exceptions as a control method is something we actively try to discourage.
* Also some clarity and consistency on how Synapse handles multiple modules implementing this callback.